lib/edit-engine/addon/routes/application.js

```diff
@@ -0,0 +1,24 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class ApplicationRoute extends Route {
+  @service content;
+
+  model(params, transition) {
+    // return whatever model the transition has
+    return Object.entries(transition.resolvedModels).reduce((acc, item) => {
+      if (item[1]) { return item[1]; }
+    }, {});
+  }
```

Comment:
Defining the routes as we have here means they are nested under their parent routes (datasets/:dataset_id / documents/:document:id). So we do not get params.dataset_id/params.document_id but we do have access to the resolved model. We also might be able to use this.paramsFor to get access to the params.